multiplayer_server/rooms/Game.php

```diff
@@ -428,7 +428,10 @@ public function finish_race($player)
     {
         global $pdo;
 
-        if ($player->finished_race === false && !isset($player->race_stats->finish_time) && $player->race_stats->drawing === false && $this->begun === true) {
+        if ($player->finished_race === false && !isset($player->race_stats->finish_time)) {
+            if($this->start_time == null) {
+                $this->start_time = 0;
```

Comment:
This is a bad default value for `start_time`, as it would cause inordinately large values for `$finish_time` in the line below. `$this -> start_time = microtime(true);` is a better way of handling this erroneous solution — after all, `start_time` shouldn't be null in the first place, so this error handler should rarely be triggered.